ddl/ddl_api.go

```diff
@@ -3984,11 +3997,17 @@ func getReplacedPartitionIDs(names []model.CIStr, pi *model.PartitionInfo) (int,
 			lastPartIdx = mathutil.Max[int](lastPartIdx, partIdx)
 		}
 	}
-	if pi.Type == model.PartitionTypeRange {
+	switch pi.Type {
+	case model.PartitionTypeRange:
 		if len(idMap) != (lastPartIdx - firstPartIdx + 1) {
 			return 0, 0, nil, errors.Trace(dbterror.ErrGeneralUnsupportedDDL.GenWithStackByArgs(
 				"REORGANIZE PARTITION of RANGE; not adjacent partitions"))
 		}
+	case model.PartitionTypeHash, model.PartitionTypeKey:
```

Comment:
Sure, let me either find or add it.

Reply:
This would need to add another ADD/COALESCE PARTITION while there is already one running, so when they are picked up from the queue, the list of partition names does no-longer match.
Let me see if I can add a test for it.

Reply:
I did not manage to create a test for it, but I with a manual test, first issue a `alter table t coalesce partiiton 1` and then when the alter started, concurrently issue a `alter table add partition partitions 1` it will hit this. If concurrently issue `alter table t coalesce partition 1` it would fail earlier in checkReorgPartitionNames.